File: 백준/Silver/10773. 제로/제로.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        InputStreamReader isr = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(isr);

        int K = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        for(int i = 0; i < K; i++){
            int N = Integer.parseInt(br.readLine());
            if(N == 0){
                stack.pop();
            } else {
                stack.push(N);
            }
        }

        int sum = 0;
        while(!stack.isEmpty()){
            sum += stack.pop();
        }

        System.out.println(sum);
    }
}

File: 백준/Silver/10773. 제로/제로.py
K = int(input())

stack = []
for i in range(K):
    n = int(input())
    if n == 0:
        stack.pop()
    else:
        stack.append(n)

if len(stack) == 0:
    print("0")
else:
    print(sum(stack))